frontend/packages/kubevirt-plugin/integration-tests/tests/dialogs/diskDialog.ts

```diff
@@ -0,0 +1,86 @@
+import { click } from '@console/shared/src/test-utils/utils';
+import { fillInput, selectOptionByText, getSelectedOptionText } from '../utils/utils';
+import * as view from '../../views/dialogs/diskDialog.view';
+import { applyButton, saveButton } from '../../views/kubevirtDetailView.view';
+import { StorageResource, DiskSourceConfig } from '../utils/types';
+import { DISK_SOURCE } from '../utils/consts';
+import { waitForNoLoaders } from '../../views/wizard.view';
+
+export class DiskDialog {
+  sourceMethods = {
+    'Attach Cloned Disk': async (sourceConfig: DiskSourceConfig) =>
+      DiskDialog.selectSourceAttachClonedDisk(sourceConfig),
+    'Attach Disk': async (sourceConfig: DiskSourceConfig) =>
+      DiskDialog.selectSourceAttachDisk(sourceConfig),
+    Container: async (sourceConfig: DiskSourceConfig) =>
+      DiskDialog.selectSourceContainer(sourceConfig),
+    URL: async (sourceConfig: DiskSourceConfig) => DiskDialog.selectSourceURL(sourceConfig),
+  };
+
+  static async selectSourceAttachDisk(sourceConfig: DiskSourceConfig) {
+    await selectOptionByText(view.diskPVC, sourceConfig.PVCName);
+  }
+
+  static async selectSourceAttachClonedDisk(sourceConfig: DiskSourceConfig) {
+    await selectOptionByText(view.diskNamespace, sourceConfig.PVCNamespace);
+    await selectOptionByText(view.diskPVC, sourceConfig.PVCName);
+  }
+
+  static async selectSourceContainer(sourceConfig: DiskSourceConfig) {
+    await selectOptionByText(view.diskContainer, sourceConfig.container);
+  }
+
+  static async selectSourceURL(sourceConfig) {
+    await selectOptionByText(view.diskURL, sourceConfig.URL);
+  }
+
+  async fillName(name: string) {
+    await fillInput(view.diskName, name);
+  }
+
+  async fillSize(size: string) {
+    if (size !== undefined && (await view.diskSize.isPresent())) {
+      await fillInput(view.diskSize, size);
+    }
+  }
+
+  async selectInterface(diskInterface: string) {
+    await selectOptionByText(view.diskInterface, diskInterface);
+  }
+
+  async selectStorageClass(storageClass: string) {
+    if (await view.diskStorageClass.isPresent()) {
+      await selectOptionByText(view.diskStorageClass, storageClass);
+    }
+  }
+
+  async getDiskSource() {
+    return getSelectedOptionText(view.diskSource);
+  }
+
+  async create(disk: StorageResource) {
+    await waitForNoLoaders();
+    if (disk.sourceConfig !== undefined) {
+      await selectOptionByText(view.diskSource, disk.sourceConfig.source);
+      await this.sourceMethods[disk.sourceConfig.source](disk.sourceConfig);
+    } else {
```

Comment:
this else doesn't have to be here if we move checking for empty config before calling the source method. 

Although I would rather and/or check if anything gets returned from the mapping.

Reply:
updated